### PR TITLE
feat(scout-for-lol): retire weekly parlay and add feature-discovery tips

### PR DESCRIPTION
**Pull request by shepherdjerred**
## Why

Retire the private-beta weekly parlay surface and make Scout's existing messages discover the features each guild has not used yet.

### Weekly parlay retirement

Per-match parlay betting (`src/betting/parlays/**`) is unchanged. The weekly market, its Temporal schedules, control surface, Discord/web UI, data models, and flag are removed. Bryan Bucks analytics sync now has its own internal endpoint, credentials, and worker activity. Weekly ledger effects are folded into auditable adjustment rows before the old tables are dropped; `BucksLedgerEntry.weeklyParlayBetId` is removed in this migration.

### Feature tips

Tips appear in existing post-match reports, pre-match alerts, and Bryan Bucks settlement DMs when the feature is enabled, available, and unused. A typed percentage roll and per-audience cooldown pace them; channel and DM audiences are independent. Tips never displace an occupied footer, add a standalone message, or combine with the settlement notification hint.

A database claim is inserted before sending. A partial unique audience index serializes concurrent claims across different tip keys. Accepted sends confirm the claim into durable history; rejected sends release it. Persisted keys are validated at the database boundary, and guild removal deletes tip impressions with the rest of guild data. The feature is gated by `feature_tips_enabled`, default off with beta targeting.

Supporting changes keep guild-specific channel decoration per send and consolidate the shared analytics capture path.

## Verification

- `bunx turbo run typecheck lint test --filter=@scout-for-lol/backend`
- `bunx lefthook run pre-commit`
- Targeted Vitest: tip eligibility/claim races and removed-guild cleanup
- Prisma generation and migration source inspection
- Buildkite exhaustive CI is required on the submitted exact head

## Not verified here

ArgoCD reconciliation, live Discord delivery, and Temporal schedule deletion require deployment/live credentials and are not performed by this PR.

### CODE REVIEW

**Comment by shepherdjerred** [comment]
This change is part of the following stack:

- #2875 ◀

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

**Comment by shepherdjerred** [comment]
### Feature tip payload, rendered through the real code path

The exact `MessageCreateOptions` Discord receives for a post-match report, before and after decoration. Produced by calling `withFeatureTip` on a realistic report message — not a mock of the output.

**Before**

```json
{
  "content": "**jerred** just finished a game — Ahri, 12/2/8, 34m, **Victory**",
  "embeds": [
    { "color": 3066993, "image": { "url": "attachment://match-report.png" } }
  ]
}
```

**After**

```json
{
  "content": "**jerred** just finished a game — Ahri, 12/2/8, 34m, **Victory**",
  "embeds": [
    {
      "color": 3066993,
      "image": { "url": "attachment://match-report.png" },
      "footer": {
        "text": "Tip: run a season-long competition with automatic daily leaderboards — set one up in the dashboard."
      }
    }
  ]
}
```

The content and image are untouched; the tip is purely additive in the footer.

Both refusals verified in the same run:

- a message whose embed footer is already used → returned unchanged (`true`)
- a plain-text message with no embed → returned unchanged (`true`)

A live Discord render needs the bot token and a real send, so it is not included here.

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `69109bf` from the CI review gate.</sub>

File: packages/scout-for-lol/packages/backend/prisma/migrations/20260913140000_remove_weekly_parlays/migration.sql [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    'weekly_parlay_stake',
+    'weekly_parlay_reserve',
+    'weekly_parlay_payout',
+    'weekly_parlay_refund',
+    'weekly_parlay_release'
+)
+GROUP BY l."bucksAccountId", a."balance"
+HAVING SUM(l."delta") <> 0;
+
+DELETE FROM "BucksLedgerEntry" WHERE "kind" IN (
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/prisma/migrations/20260913140000_remove_weekly_parlays/migration.sql]: Fixed — and you were right that documenting the pre-deploy step was not enough.

The migration now opens with a guard that aborts when any `BucksWeeklyParlayBet` is `pending` or any market is outside `settled`/`voided`. Because Prisma wraps the migration in a transaction, the `RAISE EXCEPTION` rolls everything back, so a mistimed run changes nothing rather than stranding reserved Bucks.

Exercised both ways against a scratch Postgres:

- seeded with one pending bet and one open market → `Refusing to retire weekly parlays: 1 pending bet(s) and 1 unsettled market(s) remain.`; all five tables and the bet still present afterward.
- settled market and bet → applies, tables drop, and every wallet's balance still equals its ledger sum.

File: packages/temporal/src/schedules/register-schedules.ts [hunk 2]
```diff
@@ -77,6 +77,11 @@ export const DELETED_SCHEDULE_IDS = [
   // Replaced by per-execution temporal-failure-watch alerts and worker-task
   // health guardrails. Delete the old aggregate alert on worker startup.
   "agent-task-timeout-watch",
+  // The weekly parlay feature was retired: its workflow types
+  // (`runScoutWeeklyParlayWorkflow`, `runScoutWeeklyParlayCatchupWorkflow`)
+  // are no longer in the bundle, so the live schedule must be deleted on
+  // startup rather than left firing a missing workflow.
+  "scout-weekly-parlay",
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/temporal/src/schedules/register-schedules.ts:93]: Fixed. I had not found `RETIRED_WORKFLOW_TYPES` — thanks for pointing at it.

Both `runScoutWeeklyParlayWorkflow` and `runScoutWeeklyParlayCatchupWorkflow` are on that list now, so reconciliation terminates any open execution instead of leaving it retrying against a bundle with no handler. This matters more than the usual case because a weekly execution stays open for a full week, so a deploy landing mid-run is likely rather than a corner case.

`register-schedules.test.ts` now asserts all three queries are issued, plus a dedicated test that the list covers both retired types.

File: packages/scout-for-lol/packages/backend/src/config/dynamic.ts [hunk 3]
```diff
@@ -130,6 +130,33 @@ const DEFINITION = {
...
+   * Percent of eligible messages that carry a feature tip. The roll is only
+   * reached after the audience's cooldown has expired and an unused, available
+   * tip exists, so this bounds noise within that set rather than overall.
+   */
+  featureTipPercent: {
+    schema: z.coerce.number().int().min(0).max(100),
+    sources: ["flag", "env", "default"],
+    default: 10,
+    names: {
+      flag: "scout-feature-tip-percent",
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/config/dynamic.ts:166]: Fixed. Both are declared in `managed-flag-inventory.json` as `scout-dynamic` numeric variants (defaults "10" and "72", matching the definition seeds) and `managed-flag-keys.generated.ts` is regenerated.

Worth noting this would not have surfaced locally: `dev-web-environment.ts` supplies static overrides, so the `FlagNotFoundError` and the recurring refresh warnings would only have appeared against a live Flipt.

File: packages/scout-for-lol/packages/backend/src/tips/tip-eligibility.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    (await db.tournamentLobby.count({ where: { serverId }, take: 1 })) > 0,
+};
+
+async function isAvailable(
+  tip: FeatureTip,
+  serverId: DiscordGuildId,
+): Promise<boolean> {
+  return tip.flag === "always"
+    ? true
+    : await isPolicyEnabled(tip.flag, { server: serverId });
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/tips/tip-eligibility.ts]: Fixed. `FeatureTip.flag` became `flags: readonly FlagName[]`, and availability requires every flag in the list, so a tip is only a candidate when the action behind it would actually succeed.

`dares` and `transfers` now name `["betting_enabled", <own flag>]`. A betting shutdown is exactly the moment the old behaviour would have been most misleading.

Covered by an integration test that enables the narrow flags alone (neither tip offered), then adds `betting_enabled` (both offered).

File: packages/scout-for-lol/packages/backend/src/tips/tip-eligibility.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    (await db.report.count({ where: { serverId }, take: 1 })) > 0,
+  "queue-filters": async (serverId, db) =>
+    (await db.subscription.count({
+      where: { serverId, filters: { not: null } },
+      take: 1,
+    })) > 0,
+  // One tracked player is the bare minimum a working install has; the tip is
+  // for servers that never added the rest of their group.
+  "track-more-players": async (serverId, db) =>
+    (await db.subscription.count({ where: { serverId }, take: 2 })) > 1,
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/tips/tip-eligibility.ts]: Fixed. The probe now selects with `distinct: ["playerId"]` and compares the row count, so a single player subscribed in two channels reads as one player.

I moved these tests to a real database rather than a Prisma mock, because this is a query-semantics bug a mock cannot prove. Confirmed the test fails against the previous `subscription.count` implementation and passes with the fix.

File: packages/scout-for-lol/packages/backend/prisma/migrations/20260913140000_remove_weekly_parlays/migration.sql [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+  END IF;
+END $$;
+
+INSERT INTO "BucksLedgerEntry" (
+    "bucksAccountId", "delta", "balanceAfter", "kind", "matchId", "context"
+)
+SELECT
+    l."bucksAccountId",
+    SUM(l."delta")::INTEGER,
+    a."balance",
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/prisma/migrations/20260913140000_remove_weekly_parlays/migration.sql]: Fixed — and this one changed the approach rather than patching it.

You are right that delete-and-append corrupts the history. `auditAccountLedger` walks by id and compares each stored `balanceAfter` against the running sum, so removing a delta from the middle makes every later row on that wallet drift permanently; and `loadWeeklyBucksStats` windows on `createdAt`, so the appended row would count the whole historical net as current-week movement.

The migration now **converts each row in place** to `adjustment`, leaving id, createdAt, delta and balanceAfter exactly as they were. Only the kind and its explanation change — which is also simpler than what it replaced.

Re-verified against a scratch Postgres seeded with the shape that broke: a weekly entry in the middle of a wallet's history with an ordinary `earn_game` after it. Every row's stored balanceAfter still equals the running sum, ids and order are unchanged, and no createdAt moved. Under the old approach that `earn_game` row would have derived 125 against a stored 118.

File: packages/scout-for-lol/packages/backend/src/league/tasks/notification-filters.ts [hunk 7]
```diff
@@ -188,6 +208,7 @@ export async function deliverToChannels(params: {
       }
       deliveredGuildIds.add(guildId);
       messageIdsByChannel.set(channel, sentMessage.id);
+      await decorated?.confirm();
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/league/tasks/notification-filters.ts]: Fixed. Confirmation now runs before `completeScoutEffectWithResult` rather than after.

That closes the window you describe: previously a crash between completing the effect and recording the impression left the tip delivered but unrecorded, and the retry took the `completed` branch and skipped confirmation, so the audience could be shown the same tip again.

I chose ordering over persisting the tip alongside the effect because it inverts the failure to the cheap direction: the worst case is now a recorded tip whose message is resent, costing at most one tip, instead of breaking the once-only promise the docs make.

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `0cfdc8f` from the CI review gate.</sub>

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `dd5a655` from the CI review gate.</sub>

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `02ac4f6` from the CI review gate.</sub>

File: packages/temporal/src/schedules/register-schedules.ts [hunk 8]
```diff
@@ -77,13 +77,26 @@ export const DELETED_SCHEDULE_IDS = [
...
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/temporal/src/schedules/register-schedules.ts:93]: Fixed — you were right that the previous change did nothing.

The schedule was declared `namespace: "beta"`, while `DELETED_SCHEDULES` stamped every id `prod` and reconciliation filters deletions by namespace, so the entry never matched. `terminateRetiredWorkflowExecutions` was likewise gated on `options.namespace === "prod"`. Beta would have deleted nothing and terminated nothing.

Now the namespace travels with the entry: retired schedules and retired workflow types each declare where they ran, and termination is scoped to the namespace being reconciled rather than hardcoded to prod.

Tests assert that `scout-weekly-parlay` is queued for deletion **in beta**, that a beta reconciliation queries exactly the two weekly parlay workflow types, and that a prod reconciliation still queries only `observeReviewSignalsWorkflow`.

File: packages/scout-for-lol/packages/backend/src/tips/tip-selection.ts [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    lastShown !== undefined &&
+    now.getTime() - lastShown.getTime() <
+      featureTipCooldownHours() * MS_PER_HOUR
+  ) {
+    return undefined;
+  }
+
+  const candidates = await readers.eligibleTips({
+    serverId: audience.serverId,
+    alreadyShown: await readers.shownTipKeys(audience),
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/tips/tip-selection.ts:90]: Fixed by making the impression insert the claim.

Selection now claims before the send instead of recording after it, and a unique constraint on `(serverId, audienceId, tipKey)` decides the race — exactly one concurrent delivery wins, and the loser simply carries no tip. A failed send releases the claim so the tip stays eligible.

One detail worth calling out: the column is `audienceId` with a `""` sentinel for the channel audience rather than a nullable `discordId`. Postgres treats NULLs as distinct, so a nullable column would have left every channel-scoped row outside the very constraint doing the work. It is also not named `discordId` because it holds a sentinel as well as an id.

The table is new in this PR and has never been applied anywhere, so this went into the original migration rather than stacking a second one.

Verified against a scratch Postgres that the unique index exists and a duplicate insert is rejected by name, plus tests for: two concurrent claims yielding exactly one winner, a channel claim and a DM claim for the same tip not colliding, and a released claim becoming available again.

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `ac187e3` from the CI review gate.</sub>

File: packages/scout-for-lol/packages/backend/prisma/migrations/20260913140000_remove_weekly_parlays/migration.sql [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    'weekly_parlay_release'
+);
+
+ALTER TABLE "BucksLedgerEntry" DROP CONSTRAINT IF EXISTS "BucksLedgerEntry_weeklyParlayBetId_fkey";
+
+DROP TABLE IF EXISTS "BucksWeeklyParlayDelivery";
+DROP TABLE IF EXISTS "BucksWeeklyParlayContribution";
+DROP TABLE IF EXISTS "BucksWeeklyParlayBet";
+DROP TABLE IF EXISTS "BucksWeeklyParlayMarket";
+DROP TABLE IF EXISTS "BucksWeeklyParlayDefinition";
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/prisma/migrations/20260913140000_remove_weekly_parlays/migration.sql:79]: Fixed by removing the claim rather than trying to honour it.

You are right that the two halves contradicted each other: keeping `weeklyParlayBetId` bought nothing while the same migration dropped the tables the pre-removal image reads. Rather than retain five dead tables for a release, the column is dropped too and the migration now says plainly that rolling back past it is unsupported and recovery is a restore.

What makes that acceptable is the guard directly below: the migration refuses to run while any pending bet or unsettled market exists, so the destructive steps only ever execute against a fully settled feature, after the flag is off and open markets are refunded. The PR body carries that sequencing.

The legacy importer's explicit `weeklyParlayBetId: null` is gone with it.

File: packages/scout-for-lol/packages/backend/src/tips/tip-state.ts [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+/** The tip keys this audience has already been shown. */
+export async function shownTipKeys(
+  audience: TipAudience,
+  db: ExtendedPrismaClient = prisma,
+): Promise<Set<string>> {
+  const rows = await db.featureTipImpression.findMany({
+    where: audienceWhere(audience),
+    select: { tipKey: true },
+  });
+  return new Set(rows.map((row) => row.tipKey));
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/tips/tip-state.ts]: Fixed. `shownTipKeys` now parses every stored key against the catalog at the read boundary, and an unrecognised one throws rather than joining the shown-set — where, as you note, it would have occupied the cooldown while displaying nothing.

Parsing goes through the catalog itself rather than a second hand-maintained enum, so the valid set cannot drift from the tips that exist.

Covered by a test that writes a corrupt `tipKey` straight to the table and asserts the read rejects it.

File: packages/scout-for-lol/packages/backend/src/league/tasks/notification-filters.ts [hunk 12]
```diff
@@ -174,15 +190,26 @@ export async function deliverToChannels(params: {
...
       });
+      // Before the effect is completed, not after: a crash in between would
+      // otherwise leave the tip delivered but unrecorded, and the retry would
+      // take the `completed` branch below and never confirm it — so the same
+      // audience could be shown the same tip twice. Confirming first inverts
+      // the risk to a recorded tip whose message is resent, which costs at
+      // most one tip rather than breaking the once-only promise.
+      await decorated?.confirm();
       if (effectKey !== undefined) {
         await completeScoutEffectWithResult(effectKey, sentMessage.id);
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/league/tasks/notification-filters.ts:220]: This one describes the previous revision — `confirmation()` no longer exists.

The round before last moved the claim to *before* the send: `claimTip` inserts the impression first, and if that insert fails the surrounding catch returns the message undecorated, so no tip is rendered at all. There is no longer a path where Discord accepts a tipped message whose impression was not persisted, which is what gating effect completion was meant to prevent.

What remains after the send is `confirm()`, and it now does analytics only — correctly fail-open, since a lost PostHog event must not fail a delivered message.

Resolving as addressed; happy to reopen if you see a path I have missed on the current head.

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `1073509` from the CI review gate.</sub>

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `b363cc4` from the CI review gate.</sub>

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `a2a2673` from the CI review gate.</sub>

File: packages/scout-for-lol/packages/backend/prisma/migrations/20260913150000_feature_tip_impressions/migration.sql [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+    "id" SERIAL NOT NULL,
+    "serverId" TEXT NOT NULL,
+    "audienceId" TEXT NOT NULL DEFAULT '',
+    "tipKey" TEXT NOT NULL,
+    "shownAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    CONSTRAINT "FeatureTipImpression_pkey" PRIMARY KEY ("id")
+);
+
+CREATE UNIQUE INDEX IF NOT EXISTS "FeatureTipImpression_serverId_audienceId_tipKey_key" ON "FeatureTipImpression"("serverId", "audienceId", "tipKey");
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/prisma/migrations/20260913150000_feature_tip_impressions/migration.sql:24]: Fixed in aa163fe33a: the migration now adds a partial unique index on (serverId, audienceId) for rows with claimedAt, so concurrent claims with different tip keys still serialize per audience.

File: packages/scout-for-lol/packages/backend/src/league/tasks/notification-filters.ts [hunk 14]
```diff
@@ -194,6 +225,9 @@ export async function deliverToChannels(params: {
         messageId: sentMessage.id,
       });
     } catch (error) {
+      // The decoration claimed its tip before the send; give it back so the
+      // audience stays eligible for it.
+      await decorated?.release();
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/league/tasks/notification-filters.ts]: Fixed in aa163fe33a: delivery tracks whether Discord accepted the send and releases a tip claim only when the send was not accepted, so post-send bookkeeping failures cannot make a delivered tip eligible again.

File: packages/scout-for-lol/packages/backend/src/betting/settlement/settlement-dm-delivery.ts [hunk 15]
```diff
@@ -360,6 +361,22 @@ export async function deliverSettlementDms(
   });
   for (const message of messages) {
     let status: DmStatus | undefined;
+    // A DM carries the notification hint or a feature tip, never both: two
+    // asides on one settlement receipt is the noise the DM budget exists to
+    // prevent, and the hint is the more actionable of the two.
+    const tipped = message.showHint
+      ? undefined
+      : await decorateEmbedWithFeatureTip(
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/betting/settlement/settlement-dm-delivery.ts:406]: Fixed in aa163fe33a: settlement DM delivery records the send status and releases the claim only for pre-acceptance failures; accepted sends retain the impression even if later bookkeeping fails.

File: packages/scout-for-lol/packages/backend/src/tips/tip-state.ts [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+
+/** When this audience last saw a tip, or undefined if it never has. */
+export async function lastTipShownAt(
+  audience: TipAudience,
+  db: ExtendedPrismaClient = prisma,
+): Promise<Date | undefined> {
+  const row = await db.featureTipImpression.findFirst({
+    where: audienceWhere(audience),
+    orderBy: { shownAt: "desc" },
+    select: { shownAt: true },
```
[the remark this answers is not in the document]
Comment (shepherdjerred) [reply packages/scout-for-lol/packages/backend/src/tips/tip-state.ts]: Fixed in aa163fe33a: lastTipShownAt now selects and parses every persisted tip key before reducing the newest timestamp, so malformed rows cannot suppress the cooldown silently.

**Comment by shepherdjerred** [comment]
@codex review

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `aa163fe` from the CI review gate.</sub>

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `987ccd5` from the CI review gate.</sub>

**Comment by shepherdjerred** [comment]
@codex review

<sub>Automated re-review request for `c3dba18` from the CI review gate.</sub>
